src/pymmcore_widgets/mda/_core_mda.py

```diff
@@ -0,0 +1,251 @@
+from __future__ import annotations
+
+from contextlib import suppress
+from typing import TYPE_CHECKING, cast
+
+from fonticon_mdi6 import MDI6
+from pymmcore_plus import CMMCorePlus, Keyword
+from qtpy.QtCore import QSize
+from qtpy.QtWidgets import (
+    QBoxLayout,
+    QCheckBox,
+    QFileDialog,
+    QGridLayout,
+    QGroupBox,
+    QHBoxLayout,
+    QLabel,
+    QLineEdit,
+    QPushButton,
+    QWidget,
+)
+from superqt.fonticon import icon
+
+from pymmcore_widgets.useq_widgets import MDASequenceWidget
+
+from ._core_positions import CoreConnectedPositionTable
+from ._core_z import CoreConnectedZPlanWidgert
+
+if TYPE_CHECKING:
+    from typing import TypedDict
+
+    from useq import MDASequence
+
+    class SaveInfo(TypedDict):
+        save_dir: str
+        file_name: str
+        split_positions: bool
+        should_save: bool
+
+
+class MDAWidget(MDASequenceWidget):
+    """Widget for running MDA experiments, connecting to a MMCorePlus instance."""
+
+    def __init__(
+        self, *, parent: QWidget | None = None, mmcore: CMMCorePlus | None = None
+    ) -> None:
+        # create a couple core-connected variants of the tab widgets
+        self._mmc = mmcore or CMMCorePlus.instance()
+        position_wdg = CoreConnectedPositionTable(1, self._mmc)
+        z_wdg = CoreConnectedZPlanWidgert(self._mmc)
+
+        super().__init__(parent=parent, position_wdg=position_wdg, z_wdg=z_wdg)
+
+        self.save_info = _SaveGroupBox(parent=self)
+        self.control_btns = _MDAControlButtons(self._mmc, self)
+
+        # -------- initialize -----------
+
+        self.z_plan.setSuggestedStep(_guess_NA(self._mmc) or 0.5)
+        # TODO: connect objective change event to update suggested step
+        self._update_channel_groups()
+
+        # ------------ layout ------------
+
+        layout = cast("QBoxLayout", self.layout())
+        layout.insertWidget(0, self.save_info)
+        layout.addWidget(self.control_btns)
+
+        # ------------ connect signals ------------
+
+        self.control_btns.run_btn.clicked.connect(self._on_run_clicked)
+        self.control_btns.pause_btn.released.connect(self._mmc.mda.toggle_pause)
+        self.control_btns.cancel_btn.released.connect(self._mmc.mda.cancel)
+        self._mmc.mda.events.sequenceStarted.connect(self._on_mda_started)
+        self._mmc.mda.events.sequenceFinished.connect(self._on_mda_finished)
+        self._mmc.events.channelGroupChanged.connect(self._update_channel_groups)
```

Comment:
maybe add `systemConfigurationLoaded` event?
```suggestion
        self._mmc.events.channelGroupChanged.connect(self._update_channel_groups)
        self._mmc.events.systemConfigurationLoaded.connect(self._update_channel_groups)
```